integration/src/integration/java/pl/allegro/tech/hermes/integration/AsyncPublisherAutoRetryTest.java

```diff
@@ -59,7 +59,7 @@ public void shouldNotRetryIfMessageWasPublished() {
     }
 
     private HermesClient retryingHermesClient(int retries) {
-         return HermesClientBuilder.hermesClient(new RestTemplateHermesSender(new AsyncRestTemplate()))
+         return HermesClientBuilder.hermesClient(new WebClientHermesSender(WebClient.create()))
```

Comment:
I'm not sure why we're changing this. I'd rather vote for adding separate tests for `WebClientHermesSender`. 
Even though `RestTemplateHermesSender` is deprecated, we're still maintaining it, so it would be nice to have tests to see if it works properly.